src/Lean/Elab/Command.lean

```diff
@@ -248,19 +265,8 @@ register_builtin_option showPartialSyntaxErrors : Bool := {
 builtin_initialize registerTraceClass `Elab.command
 
 partial def elabCommand (stx : Syntax) : CommandElabM Unit := do
-  let mkInfoTree trees := do
-    let ctx ← read
-    let s ← get
-    let scope := s.scopes.head!
-    let tree := InfoTree.node (Info.ofCommandInfo { stx := stx }) trees
-    let tree := InfoTree.context {
-      env := s.env, fileMap := ctx.fileMap, mctx := {}, currNamespace := scope.currNamespace, openDecls := scope.openDecls, options := scope.opts
-    } tree
-    if checkTraceOption (← getOptions) `Elab.info then
-      logTrace `Elab.info m!"{← tree.format}"
-    return tree
   let initMsgs ← modifyGet fun st => (st.messages, { st with messages := {} })
-  withLogging <| withRef stx <| withInfoTreeContext (mkInfoTree := mkInfoTree) <| withIncRecDepth <| withFreshMacroScope do
```

Comment:
Why did you have to move `withInfoTreeContext`?

Reply:
Is it because you need to set the "elaborator"?

Reply:
Yes, I needed to move all info tree construction to where that information is known